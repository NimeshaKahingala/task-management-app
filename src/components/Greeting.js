import React from "react";

function Greeting(props) {
    if(!props.name) {
        throw new Error("Name is required")
    }
    return(
        <h2>hello {props.name}</h2>
    )
}

export default Greeting;