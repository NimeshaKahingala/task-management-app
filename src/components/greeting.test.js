import Greeting from "./Greeting";
import { render } from "@testing-library/react";

describe("greeting", ()=>{
    test("greeting component render correctly",()=>{
        const { asFragment} = render(<Greeting name="Lisa"/>);
        expect(asFragment()).toMatchSnapshot();
    })
 
    test("throws error when name is not provided",()=>{
     expect(()=> render(<Greeting/>)).toThrow("Name is required")
    })
})