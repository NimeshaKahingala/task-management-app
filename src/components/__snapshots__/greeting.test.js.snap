// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`greeting greeting component render correctly 1`] = `
<DocumentFragment>
  <h2>
    hello Lisa
  </h2>
</DocumentFragment>
`;
